silkrpc/commands/debug_api.cpp

```diff
@@ -128,8 +128,8 @@ boost::asio::awaitable<void> DebugRpcApi::handle_debug_get_modified_accounts_by_
 
     try {
         ethdb::TransactionDatabase tx_database{*tx};
-        const auto start_block_number = co_await core::get_block_number(start_block_id, tx_database);
-        const auto end_block_number = co_await core::get_block_number(end_block_id, tx_database);
+        const auto [start_block_number, _] = co_await core::get_block_number(start_block_id, tx_database, /*latest_required=*/false);
+        const auto [end_block_number, ignore] = co_await core::get_block_number(end_block_id, tx_database, /*latest_required=*/false);
```

Comment:
I've just now realized that we should also keep the previous `get_block_number` overload:
```
awaitable<uint64_t> get_block_number(const std::string& block_id, const DatabaseReader& reader);
```
and simply delegate to the new overload in its implementation:
```
awaitable<uint64_t> get_block_number(const std::string& block_id, const DatabaseReader& db_reader) {
    const auto [block_number, _] = co_await get_block_number(block_id, db_reader, /*latest_required=*/false);
    return block_number;
}
```
This would allow us to use the first overload version wherever we don't case about `is_latest` flag (i.e. where we currently pass `/*latest_required=*/false` and ignore the returned flag), making the code shorter and clearer at call site.

Reply:
ok